src/utils/references.ts

```diff
@@ -44,3 +45,47 @@ export function stringifyReferences (paths: string[], cwd: string): string {
 export function toReference (path: string, cwd: string): string {
   return `/// <reference path="${normalizeSlashes(isAbsolute(path) ? relative(cwd, path) : normalize(path))}" />`
 }
+
+export enum DependencyTarget {
+  Main,
+  Browser
+}
+
+export enum DependencyType {
+  Ambient,
+  External
+}
+
+export interface DependencyInfo {
+  target: DependencyTarget
+  type: DependencyType
+  name: string
+}
+
+export function parseReferencePath(path: string): DependencyInfo {
+  const referencePathRegexPattern =
```

Comment:
Excellent points. Yes, literal path here was a bit short sighted. good to know about NPM packages, but is the '/' actually part of the dependency name? I.e. is it included as the key to "config.devDependencies"? Assuming yes.